### PR TITLE
[Onboarding Flow]- Write test after the "Generate Chaincode" button is clicked 

### PR DESCRIPTION
**Pull request by shubhamsigdar1**
### What is the change?

- Before fetching an API in an component, but Now I moved it to the controller file
- Fix all the test and skip test
- under feature flag

### Is Development Tested?

- [x] Yes
- [ ] No

### Test Stats

https://github.com/Real-Dev-Squad/website-www/assets/107163260/63ed8190-d72e-43fb-9a3d-a68b83982d99

### After Change Video

https://github.com/Real-Dev-Squad/website-www/assets/107163260/88215f45-cce8-4515-a7c2-b215840c99c2

### CODE REVIEW

File: app/controllers/join.js [hunk 1]
```diff
@@ -1,14 +1,63 @@
...
+import { APPS } from '../constants/urls';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { toastNotificationTimeoutOptions } from '../constants/toast-notification';
 
 export default class JoinController extends Controller {
   @service router;
   @service login;
   @service featureFlag;
+  @tracked Chaincode = 'Generate Chaincode';
```
Comment (satyam73) [inline app/controllers/join.js:12]: Why `c` is in capital, it doesn't follow the convention of the app

File: app/controllers/join.js [hunk 2]
```diff
@@ -1,14 +1,63 @@
...
+          '',
+          toastNotificationTimeoutOptions
+        );
+      } else {
+        this.toast.error(
+          'Something went wrong. Please check console errors.',
+          '',
+          toastNotificationTimeoutOptions
+        );
+      }
```
Comment (satyam73) [inline app/controllers/join.js:52]:
```suggestion
      if (!response.ok) return  this.toast.error(
          'Something went wrong. Please check console errors.',
          '',
          toastNotificationTimeoutOptions
        );
        
        this.Chaincode = chaincode;
        this.isChaincodeClicked = true;
        this.toast.info(
          'Generated New Chaincode!!',
          '',
          toastNotificationTimeoutOptions
        );
   
```

File: tests/integration/components/identity-steps/step-four-test.js
Comment (satyam73) [inline tests/integration/components/identity-steps/step-four-test.js:1]: is the test for `handleGenerateChaincode` function written  in any previous PRs

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: tests/integration/components/identity-steps/step-four-test.js
Comment (shubhamsigdar1) [reply tests/integration/components/identity-steps/step-four-test.js:1]: Didn't understand this comment?? What you want me to achieve

File: app/controllers/join.js (hunk 1 above)
Comment (shubhamsigdar1) [reply app/controllers/join.js]: Done With changes

File: app/controllers/join.js (hunk 2 above)
Comment (shubhamsigdar1) [reply app/controllers/join.js]: Done With changes

File: tests/integration/components/identity-steps/step-four-test.js
Comment (shubhamsigdar1) [reply tests/integration/components/identity-steps/step-four-test.js:1]: yes I already did test

Comment (satyam73) [reply tests/integration/components/identity-steps/step-four-test.js:1]: okay resolving this one

**Review by satyam73 — APPROVED** [review]
LGTM

**Review by gauravsinhaweb — APPROVED** [review]

**Pull request merged**